instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -33,98 +36,204 @@ type gRPCContextKey struct{}
 type gRPCContext struct {
 	messagesReceived int64
 	messagesSent     int64
+	requestSize      int64
+	responseSize     int64
+	metricAttrs      []attribute.KeyValue
 }
 
 // NewServerHandler creates a stats.Handler for gRPC server.
 func NewServerHandler(opts ...Option) stats.Handler {
 	h := &serverHandler{
 		config: newConfig(opts),
+		r:      &handler{},
 	}
 
-	h.tracer = h.config.TracerProvider.Tracer(
+	h.r.tracer = h.config.TracerProvider.Tracer(
 		instrumentationName,
 		trace.WithInstrumentationVersion(SemVersion()),
 	)
+	h.r.meter = h.config.MeterProvider.Meter(
+		instrumentationName,
+		metric.WithInstrumentationVersion(Version()),
+		metric.WithSchemaURL(semconv.SchemaURL),
+	)
+	var err error
+	h.r.rpcDuration, err = h.r.meter.Int64Histogram("rpc.server.duration",
+		metric.WithDescription("Measures the duration of inbound RPC."),
+		metric.WithUnit("ms"))
+	if err != nil {
+		otel.Handle(err)
+	}
+
+	h.r.rpcRequestSize, err = h.r.meter.Int64Histogram("rpc.server.request.size",
+		metric.WithDescription("Measures size of RPC request messages (uncompressed)."),
+		metric.WithUnit("By"))
+	if err != nil {
+		otel.Handle(err)
+	}
+
+	h.r.rpcResponseSize, err = h.r.meter.Int64Histogram("rpc.server.response.size",
+		metric.WithDescription("Measures size of RPC response messages (uncompressed)."),
+		metric.WithUnit("By"))
+	if err != nil {
+		otel.Handle(err)
+	}
+
+	h.r.rpcRequestsPerRPC, err = h.r.meter.Int64Histogram("rpc.server.requests_per_rpc",
```

Comment:
Curious, why does here use Histogram instead of Counter?

Reply:
It is following the specification: https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/rpc-metrics/#rpc-server